pkg/build/api/types.go

```diff
@@ -76,8 +76,72 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails
+
+	// CIJobDetails is a generic placeholder that should store information about
+	// jobs from CI plugins that integrate with openshift and triggered a
+	// specific build
+	CIDetails CIJobDetails
+
+	// GitWebHookInfo holds information about a build that was triggered by a
+	// gitwebhook including its refs, branch, last commit id, etc. It also stores
+	// the type of plugin that triggered it, if it was a generic one, github,
+	// bitbucket, etc.
+	GitWebHook GitWebHookInfo
+
+	// ImageChangeBuildInfo - Stores info about the image that triggered a new
+	// build
+	ImageChange ImageChangeBuildInfo
+
+	// ConfigChangeBuildInfo - stores info about the last version of the config
+	// that triggered this build
+	ConfigChange ConfigChangeBuildInfo
+}
+
+// BuildTriggerDetails holds information about who/what triggered a build
+type BuildTriggerDetails string
+
+// CIJobDetails holds information about the CI job that triggered a
+// build
+type CIJobDetails struct {
+	// Type is type of the CI that started this. "jenkins", "travis", etc.
+	Type string
+
+	// JobURL holds the url that points to the job that triggered a specific
+	// build
+	JobURL string
 }
 
+// GitWebHookInfo holds information about a git webhook that triggered a build
+type GitWebHookInfo struct {
+	// GitTriggerType is the git webhook trigger type (github, generic, etc)
+	GitTriggerType BuildTriggerType
```

Comment:
just make this struct unique to GitHub webhooks and create a separate one for GenericWebhooks.  

Otherwise you're using the word "Git" but the webhook might not have anything to do with Git.